### PR TITLE
[skip ci] Unified OVA installer test

### LINKED ISSUE #4808: Unified OVA Test v1.2
User statement:  As an engineer and customer of VIC, I want to know that the unified installer OVA has gone through at least a minimal set of testing so as to be reliable and known to work in some scenarios

Details:

Acceptance Criteria:
1. Tests are documented in .md files in the repo
2. Tests are implemented in robot files in the repo

### PR DESCRIPTION
**Pull request by rajanashok**
Initial implementation of the unified OVA tests. This merge includes the following capabilities.
- Deploys the unified OVA
- Gets the HARBOR-IP to use with VCH install
- Deploys VIC appliance with --registry-ca %{HARBOR-IP}
- Logs in to HARBOR UI using Selenium2Library
- Creates test project and users in harbor
- Runs the docker workflow pointing to the harbor instance

Fixes #4808

### CODE REVIEW

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+#Purpose:
+To verify the VIC Product (Engine/Harbor) work using insecure registry with a VC6.5 and ESX6.5 server
+
+#References:
+[1 - VIC+Harbor Integration Test Plan](https://confluence.eng.vmware.com/pages/viewpage.action?spaceKey=corevc&title=VIC+-+Harbor+Integration+Test+Plan)
+
+#Environment:
+* This test requires access to VMWare Nimbus cluster for dynamic ESXi and vCenter creation
+* Login to Harbor using an LDAP user1 (say admin role)
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md]: I would name the user `admin` if admin privileges are required to perform the actions done by this user

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+* Create a Project named say 'vic-harbor' (publicity off)
+* Login with 2 other LDAP users (say user2, user3) and logout - Limitation (Issue#)
+* Login as user1 and add other users with different roles (developer, guest) under Project 'vic-harbor'
+* Prepare 3 windows client machines (3rd client machine could be a Linux machine as we couldn't figure out the VT-X issue for running docker on Windows 7/10 VM)
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md]: Why is Windows required?

**Comment by andrewtchin** [comment]
#4808

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+*** Settings ***
+Documentation  Test 19-5 - Configuration 1 VC6.5 ESX6.5 VIC OVA Insecured
+Resource  ../../resources/Util.robot
+Suite Setup  19-1-Setup
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot]: Rename to 19-5 or a descriptive keyword name

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+Test 19-5 - Configuration 1 VC6.5 ESX6.5 VIC OVA Insecured
+=======
+
+#Purpose:
+To verify the VIC Product (Engine/Harbor) work using insecure registry with a VC6.5 and ESX6.5 server
+
+#References:
+[1 - VIC+Harbor Integration Test Plan](https://confluence.eng.vmware.com/pages/viewpage.action?spaceKey=corevc&title=VIC+-+Harbor+Integration+Test+Plan)
+
+#Environment:
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md]: I think the setup should include deploying the OVA

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md (hunk 2 above)
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md]: Also, why are 3 client machines required? I don't see anything in the `.robot` using different clients

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    Create Project And Three Users For It  developer=${developer}  developer2=${developer2}  developerEmail=${developerEmail}  developerEmail2=${developerEmail2}  developerFullName=${developerFullName}  password=${password}  comments=${comments}  guest=${guest}  developerRole=${developerRole}  guestRole=${guestRole}  project=${project}  public=${False}
+    Remove Environment Variable  DOCKER_HOST
+
+19-1-Teardown
+    Run Keyword And Continue On Failure  Cleanup VIC Appliance On Test Server
+    ${out}=  Run Keyword And Continue On Failure  Run  govc vm.destroy 19-1-harbor
+    Run Keyword And Continue On Failure  Nimbus Cleanup  ${list}  ${true}  ${true}
+
+*** Test Cases ***
+Test Pos001 Admin Operations
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot]: Please include the descriptive names for each test case in the `.md` too

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+    Wait Until Keyword Succeeds  10x  6s  Check That Datastore Is Cleaned  ${container_name}
+
+    # Docker delete image
+    Log To Console  docker rmi VCH2...
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${VCH2-URL} rmi -f %{HARBOR_IP}/${project}/${image}
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Untagged
+
+# Test Pos004 Three Client Machines With One Harbor
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot]: Is this a `TODO`? If it's something we plan to do later you can use `Get State Of Github Issue` and `Pass Execution` until this is implemented

File: tests/resources/OVA-Util.robot [hunk 7]
```diff
@@ -42,27 +42,54 @@ ${tls_not_disabled}  False
...
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc host.esxcli -host.ip=${esx1-ip} system settings advanced set -o /Net/GuestIPHack -i 1
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc host.esxcli -host.ip=${esx2-ip} system settings advanced set -o /Net/GuestIPHack -i 1
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc host.esxcli -host.ip=${esx3-ip} system settings advanced set -o /Net/GuestIPHack -i 1
+
+    ${ip}=  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc vm.ip -esxcli vic-unified-ova-integration-test
 
-    Log To Console  \nUnified OVA is deployed successfully
+    Set Environment Variable  HARBOR_IP  ${ip}
+    
+    Log To Console  \n%{HARBOR_IP}
```
Comment (andrewtchin) [inline tests/resources/OVA-Util.robot]: Log To Console  \nHarbor IP: %{HARBOR_IP}

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+*** Test Cases ***
+Test Pos001 Admin Operations
+    Basic Docker Command With Harbor  user=admin  password=%{TEST_PASSWORD}  project=${project}  image=${image}  container_name=${container_name}
+
+Test Pos002 Developer Operations
+    Basic Docker Command With Harbor  user=${developer}  password=${password}  project=${project}  image=${image}  container_name=${container_name}
+    
+Test Neg001 Developer Operations
+    # Docker login
+    Log To Console  \nRunning docker login admin...
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot]: Should this be `Running docker login guest`?

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+1. Reuse the VCH created in test Pos001
+2. Do docker login to harbor using an user with developer role
+3. Pull an image/application into the local docker registry using local docker in the client machine
+4. Tag the image/application to push it to Harbor private registry using local docker in the client machine
+5. Push the image/application from local to Harbor Private registry
+6. From VCH host, Pull the image from Harbor registry
+7. Run the application using docker run command
+8. Stop, Start, attach, kill, rm and rmi
+
+###neg001:
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md]: How is this a negative test case?

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+    # Docker login VCH2
+    Log To Console  \nRunning docker login developer2 VCH2...
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${VCH2-URL} login -u ${developer2} -p ${password} %{HARBOR_IP}
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Login Succeeded
+    Should Not Contain  ${output}  Error response from daemon
+
+    # Docker pull from dockerhub
+    Log To Console  docker pull from dockerhub...
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot]: This isn't pulling from dockerhub right?

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+#Environment:
+* This test requires access to VMWare Nimbus cluster for dynamic ESXi and vCenter creation
+* Login to Harbor using an LDAP user1 (say admin role)
+* Create a Project named say 'vic-harbor' (publicity off)
+* Login with 2 other LDAP users (say user2, user3) and logout - Limitation (Issue#)
+* Login as user1 and add other users with different roles (developer, guest) under Project 'vic-harbor'
+* Prepare 3 windows client machines (3rd client machine could be a Linux machine as we couldn't figure out the VT-X issue for running docker on Windows 7/10 VM)
+
+#Test Steps:
+###pos001:
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.md]: Please give each test case a descriptive name

File: tests/resources/Harbor-Util.robot [hunk 12]
```diff
@@ -26,7 +26,7 @@ ${harbor_cert}  getcert
 *** Keywords ***
 Install Harbor To Test Server
     [Tags]  secret
-    [Arguments]  ${name}=harbor  ${protocol}=http  ${verify}=off  ${db_password}=%{TEST_PASSWORD}  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL_ARRAY}  ${datastore}=%{TEST_DATASTORE}  ${network}=VM Network
+    [Arguments]  ${name}=harbor  ${protocol}=https  ${verify}=off  ${db_password}=%{TEST_PASSWORD}  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL_ARRAY}  ${datastore}=%{TEST_DATASTORE}  ${network}=VM Network
     Log To Console  \nFetching harbor ova...
     ${status}  ${message}=  Run Keyword And Ignore Error  OperatingSystem.File Should Exist  ${HARBOR_VERSION}.ova
     ${out}=  Run Keyword If  '${status}' == 'FAIL'  Run  wget https://github.com/vmware/harbor/releases/download/${HARBOR_SHORT_VERSION}/${HARBOR_VERSION}.ova
```
Comment (andrewtchin) [inline tests/resources/Harbor-Util.robot]: This test case can be updated to use the standalone installer or eliminated if we use the OVA installer since that would already bring up Harbor

File: tests/resources/OVA-Util.robot [hunk 13]
```diff
@@ -42,27 +42,54 @@ ${tls_not_disabled}  False
 *** Keywords ***
 # Requires vc credential for govc
 Deploy VIC-OVA To Test Server
-    [Arguments]  ${dhcp}=False  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
+    [Arguments]  ${dhcp}=True  ${protocol}=https  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL_ARRAY}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
     Run Keyword if  ${build}  Build Unified OVA
     ${rev}=  Run  git rev-parse --short HEAD
-    Set Test Variable  ${ova_path}  bin/vic-1.1.0-${rev}.ova
+    Set Global Variable  ${ova_path}  bin/vic-v1.1.0-${rev}.ova
```
Comment (andrewtchin) [inline tests/resources/OVA-Util.robot]: I think this can be moved to the top so it's easy to find when we update the version

File: tests/resources/OVA-Util.robot [hunk 14]
```diff
@@ -42,27 +42,54 @@ ${tls_not_disabled}  False
 *** Keywords ***
 # Requires vc credential for govc
 Deploy VIC-OVA To Test Server
-    [Arguments]  ${dhcp}=False  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
-    Run Keyword if  ${build}  Build Unified OVA
-    ${rev}=  Run  git rev-parse --short HEAD
-    Set Test Variable  ${ova_path}  bin/vic-1.1.0-${rev}.ova
+    [Arguments]  ${dhcp}=True  ${protocol}=https  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL_ARRAY}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
+    Set Global Variable  ${ova_path}  bin/vic-v1.1.0-bf760ea2.ova
```
Comment (rogeliosanchez) [inline tests/resources/OVA-Util.robot]: Just wanted to know if we really want to hardcode the `rev` part of the ova_path here.  What is the reason we want to do this here now?  `bin/vic-v1.1.0-bf760ea2.ova`

Comment (rajanashok) [inline tests/resources/OVA-Util.robot]: This is just for my local testing, will be using the ${rev} when merging the PR.

Comment (rogeliosanchez) [inline tests/resources/OVA-Util.robot]: Ok, just wondering.

File: tests/resources/OVA-Util.robot [hunk 15]
```diff
@@ -42,27 +42,54 @@ ${tls_not_disabled}  False
...
-    Run Keyword If  ${dhcp}  Log To Console  ovftool --datastore=${datastore} ${ova_options} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
-    ...  ELSE  Log To Console  ovftool --datastore=${datastore} ${ova_options_with_network} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
-    ${out}=  Run Keyword If  ${dhcp}  Run  ovftool --datastore=${datastore} ${ova_options} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
-    ...  ELSE  Run  ovftool --datastore=${datastore} ${ova_options_with_network} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
-    Log  ${out}
+    Run Keyword If  ${dhcp}  Log To Console  ovftool --datastore=${datastore} ${ova_options} ${ova_path} 'vi://${user}:${password}@${host}/${cluster}'
+    ...  ELSE  Log To Console  ovftool --datastore=${datastore} ${ova_options_with_network} ${ova_path} 'vi://${user}:${password}@${host}/${cluster}'
+    ${out}=  Run Keyword If  ${dhcp}  Run  ovftool --datastore=${datastore} ${ova_options} --net:"Network"="VM Network" ${ova_path} 'vi://${user}:${password}@${host}/${cluster}'
```
Comment (rogeliosanchez) [inline tests/resources/OVA-Util.robot]: Question: Can we use both single and double quotes to describe/state a string or is there a specific case to use either or?  `--net:"Network"="VM Network"`  is there a reason we use double quotes here?

File: tests/resources/OVA-Util.robot [hunk 16]
```diff
@@ -42,27 +42,54 @@ ${tls_not_disabled}  False
...
+    ${idx}=  Set Variable  1
+    :FOR  ${line}  IN  @{out}
+    \   Continue For Loop If  '${line}' == '/ha-datacenter/host/cls/Resources'
+    \   ${ip}=  Fetch From Right  ${line}  /
+    \   Set Suite Variable  ${esx${idx}-ip}  ${ip}
+    \   ${idx}=  Evaluate  ${idx}+1
+
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc host.esxcli -host.ip=${esx1-ip} system settings advanced set -o /Net/GuestIPHack -i 1
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc host.esxcli -host.ip=${esx2-ip} system settings advanced set -o /Net/GuestIPHack -i 1
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc host.esxcli -host.ip=${esx3-ip} system settings advanced set -o /Net/GuestIPHack -i 1
```
Comment (rogeliosanchez) [inline tests/resources/OVA-Util.robot]: Do we care if we have less than 3 esxs here or more since we are only checking for esx1, 2, and 3?

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+${password}  harbor-admin-passwd
+${newPassword}  Test-new-password-1
+${project}  vic-harbor
+${image}  busybox
+${container_name}  busy
+${tag}  new
+${developerRole}  Developer
+${guestRole}  Guest
+${developerEmail}  developer@test.com
+${developerEmail2}  develope2@test.com
```
Comment (rogeliosanchez) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Insecured.robot]: Nit picking here: should this be `developer2@test.com`?

File: tests/resources/Harbor-Util.robot [hunk 18]
```diff
@@ -20,13 +20,14 @@ Library  OperatingSystem
 *** Variables ***
 ${HARBOR_SHORT_VERSION}  0.5.0
 ${HARBOR_VERSION}  harbor_0.5.0-9e4c90e
-${MY_PROJECTS_TABLE}  div.table-body-container > table.table.table-pane
+${MY_PROJECTS_TABLE}  harbor-app > harbor-shell > clr-main-container > div > div > project > div > div > list-project
```
Comment (rogeliosanchez) [inline tests/resources/Harbor-Util.robot]: We may want to ask the Admiral team (or whoever does the web UI) to include/create an id for this table.  That will help shorten this css identifier quite a bit and would be beneficial when testing the UI.
Also, `div>list-project` as an identifier works instead of the longer identifier.

File: tests/resources/Harbor-Util.robot [hunk 19]
```diff
@@ -20,13 +20,14 @@ Library  OperatingSystem
...
+${MY_MEMBERS_TABLE}  body > harbor-app > harbor-shell > clr-main-container > div > div > project-detail > ng-component > div > div:nth-child(2) > clr-datagrid > div > div > div > div.datagrid-body
```
Comment (rogeliosanchez) [inline tests/resources/Harbor-Util.robot]: `div.datagrid-body` also works as an identifier here instead of the longer one.

File: tests/resources/Harbor-Util.robot [hunk 20]
```diff
@@ -82,68 +83,65 @@ Install Harbor Self Signed Cert
...
-    Input Text  uPassword  ${pw}
-    Click button  Sign In
-    Wait Until Page Contains  Summary
-    Wait Until Page Contains  My Projects:
-    Wait Until Keyword Succeeds  5x  1  Page Should Contain Element  xpath=//optional-menu/div/a[contains(., '${user}')]
+    Input Text  login_username  ${user}
+    Input Text  login_password  ${pw}
+    Click button  LOG IN
+    Wait Until Page Contains  Projects
+    Wait Until Keyword Succeeds  5x  1  Page Should Contain Element  xpath=/html/body/harbor-app/harbor-shell/clr-main-container[@class='main-container']/navigator/clr-header[@class='header-5 header']/div[@class='header-actions']/clr-dropdown[@class='dropdown bottom-right']/button[@class='nav-text dropdown-toggle']/span[contains(., '${user}')]
```
Comment (rogeliosanchez) [inline tests/resources/Harbor-Util.robot]: Can you try `clr-dropdown>button>span` as an identifier here?

Comment (rajanashok) [inline tests/resources/Harbor-Util.robot]: @rogeliosanchez These are valid concerns, but I think these can be part of subsequent PR merges, as there is still more UI work for other test cases. I also want to have the UI change/test cycle to a minimum at this point.

In the meantime, it would be good to create a story for tagging IDs so we could track this with the Admiral team. Can you please create one and tag it to this PR?

File: tests/resources/OVA-Util.robot (hunk 16 above)
Comment (rajanashok) [inline tests/resources/OVA-Util.robot]: We are using a Simple VC Cluster here that has 3 esx's.

File: tests/resources/Harbor-Util.robot [hunk 21]
```diff
@@ -20,48 +20,12 @@ Library  OperatingSystem
...
 ${HARBOR_SHORT_VERSION}  0.5.0
 ${HARBOR_VERSION}  harbor_0.5.0-9e4c90e
-${MY_PROJECTS_TABLE}  div.table-body-container > table.table.table-pane
+${MY_PROJECTS_TABLE}  harbor-app > harbor-shell > clr-main-container > div > div > project > div > div > list-project
+${MY_MEMBERS_TABLE}  body > harbor-app > harbor-shell > clr-main-container > div > div > project-detail > ng-component > div > div:nth-child(2) > clr-datagrid > div > div > div > div.datagrid-body
 ${harbor_cert}  getcert
+${ova_harbor_admin_password}  harbor-admin-passwd
 
 *** Keywords ***
-Install Harbor To Test Server
```
Comment (andrewtchin) [inline tests/resources/Harbor-Util.robot]: Even though this uses the outdated installer it needs to be kept for now since there are other tests that use it

File: tests/resources/OVA-Util.robot [hunk 22]
```diff
@@ -42,27 +42,54 @@ ${tls_not_disabled}  False
 *** Keywords ***
 # Requires vc credential for govc
 Deploy VIC-OVA To Test Server
-    [Arguments]  ${dhcp}=False  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
+    [Arguments]  ${dhcp}=True  ${protocol}=http  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL_ARRAY}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
     Run Keyword if  ${build}  Build Unified OVA
     ${rev}=  Run  git rev-parse --short HEAD
-    Set Test Variable  ${ova_path}  bin/vic-1.1.0-${rev}.ova
+    Set Global Variable  ${ova_path}  bin/vic-1.1.0-${rev}.ova
```
Comment (andrewtchin) [inline tests/resources/OVA-Util.robot]: Please put this at the top of the file or its own keyword so it's easy to find to update. The 1.1.0 part is probably already outdated.

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Secured-SelfSignedCACert.md [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+2. Using different machines, connect to the same VCH (in parallel)
+3. Do docker login to harbor using different users (in parallel)
+4. Pull an image/application into the local docker registry using local docker in the client machine
+5. Tag the image/application to push it to Harbor private registry using local docker in the client machine
+6. Push the image/application from local to Harbor Private registry
+7. From VCH, Pull same/different images using different users from Harbor registry
+8. Run the applications using different users from Harbor registry (in parallel)
+9. Stop, Start, attach, kill, rm and rmi using different users from Harbor registry (in parallel)
+
+#Expected Outcome:
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Secured-SelfSignedCACert.md]: Nit, but could you put the expected outcome of each test with each test case step description?

File: tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Secured-SelfSignedCACert.robot [hunk 24]
```diff
@@ -0,0 +1,232 @@
...
+    Wait Until Keyword Succeeds  10x  6s  Check That Datastore Is Cleaned  ${container_name}
+
+    # Docker delete image
+    Log To Console  docker rmi VCH2...
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${VCH2-URL} rmi -f %{HARBOR_IP}/${project}/${image}
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Untagged
+
+# Test Pos004 Three Client Machines With One Harbor
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-5-Configuration-1-VC6.5-ESX6.5-VIC-OVA-Harbor-Secured-SelfSignedCACert.robot]: Why isn't this one implemented? Put `Pass Execution . Not implemented because <reason>`

**Comment by andrewtchin** [comment]
Looks good after above comments are addressed

**Comment by rajanashok** [comment]
@andrewtchin Please review, updated the PR with the review comments.

**Comment by andrewtchin** [comment]
need to uncomment line 231 so that's a running test case, otherwise lgtm

**Comment by rajanashok** [comment]
This merge will handles the following.
-Deploys the unified OVA
-Gets the HARBOR-IP to use with VCH install
-Deploys VIC appliance with --registry-ca %{HARBOR-IP}
-Logs in to HARBOR UI using Selenium2Library
-Creates test project and users in harbor
-Runs the docker workflow pointing to the harbor instance

Known issues to be addressed in the following sprint.
- All test users created in Harbor are with admin role, need to fix it to set developer and guest role
- All Docker commands for admin operations against Harbor passing except, the tests fails during Checking for VM Removal on Keyword 'Check That VM Is Removed'
